operator/pkg/manifest/crd.go

```diff
@@ -0,0 +1,95 @@
+package manifest
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"io"
+	"io/ioutil"
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/apimachinery/pkg/util/yaml"
+	"os"
+	"path/filepath"
+)
+
+func GetCRDsFromPath(filePath string) ([]*apiextensionsv1.CustomResourceDefinition, error) {
+	pathInfo, err := os.Stat(filePath)
+	if os.IsNotExist(err) {
+		return nil, nil
+	} else if err != nil {
+		return nil, err
+	}
+
+	var files []os.FileInfo
+	if !pathInfo.IsDir() {
+		filePath, files = filepath.Dir(filePath), []os.FileInfo{pathInfo}
+	} else if files, err = ioutil.ReadDir(filePath); err != nil {
+		return nil, err
+	}
+
+	crdsList, err := readCRDs(filePath, files)
+	if err != nil {
+		return nil, err
+	}
+
+	return crdsList, nil
+}
+
+func readCRDs(basePath string, files []os.FileInfo) ([]*apiextensionsv1.CustomResourceDefinition, error) {
```

Comment:
this method is heavy :O I would suggest switching to the apiextensions yaml conversion which also supports json etc. out of the box.

Reply:
it could be heavy yes, if used frequently. Once in a lifetime usage should be find. On top I have added other extensions as well. 
Also, I didn#t find any yaml->CRD conversion method out of the box :/

Reply:
```go
import "k8s.io/apimachinery/pkg/util/yaml"
import "os"
import v1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
func ReadFromFileIntoCRD(path string) *v1.CustomResourceDefinition {
        crd := v1.CustomResourceDefinition{}
	file, err := os.Open(filepath.Clean(path))
	if err != nil {
		return nil, err
	}
	err = yaml.NewYAMLOrJSONDecoder(file, 2048).Decode(&crd)
	if err != nil {
		return nil, err
	}
	return &crd, nil
}